src/schema/partner_show_event.ts

```diff
@@ -1,8 +1,46 @@
-import date from "./fields/date"
-import { GraphQLString, GraphQLObjectType } from "graphql"
+import dateField, { date, DateSource } from "./fields/date"
+import { GraphQLString, GraphQLObjectType, GraphQLFieldConfig } from "graphql"
 import { ResolverContext } from "types/graphql"
 import { exhibitionPeriod } from "lib/date"
 
+const checkUserAgent = (userAgent): boolean =>
+  userAgent!.indexOf("Artsy-Mobile/4") > 0 ||
+  userAgent!.indexOf("Artsy-Mobile/5.0.0") > 0 ||
+  userAgent!.indexOf("Artsy-Mobile/5.0.1") > 0
+
+const isOlderEmissionVersion = (userAgent): boolean => {
+  let result = false
+  if (typeof userAgent === "string") {
+    result = checkUserAgent(userAgent)
+  } else if (Array.isArray(userAgent)) {
+    result = userAgent.some(ua => checkUserAgent(ua))
+  }
+  return result
+}
+
+const dateFieldForPartnerShowEvent: GraphQLFieldConfig<
+  DateSource,
+  ResolverContext
+> = {
+  ...dateField,
```

Comment:
This reuses the existing field definition, and only provides this custom resolver below.